src/lib.rs

```diff
@@ -51,7 +51,12 @@
 #![allow(
     clippy::use_self,
     clippy::single_match_else,
-    clippy::must_use_candidate
+    clippy::must_use_candidate,
+
+    // temporary clearing some clippys for dev
```

Comment:
Add a big TODO flag here please, there is a big chance we'll overlook it before merging.

Reply:
I've added a check-box linking to this comment. Made the changes already, and will go into the next commit.